GSoC-Ideas.md

```diff
@@ -1,18 +1,27 @@
 # Ideas for Google Summer of Code projects
 
-### Idea #x:
+### Idea #x: Support of Source Code Related Metrics
 
 [ Micro-tasks and place for questions ](#)
 
-Description
+Currently, GrimoireLab allows to produce analytics with data extracted from more than 30 tools related with contributing to Open Source development such as version control systems, issue trackers and forums. Despite the large set of metrics available in GrimoireLab, none of them relies on information extracted from source code, thus limiting the end-users to benefit of a wider spectrum of software development data.
+
+[Graal](https://github.com/Bitergia/graal) is a tool that allows to conduct customizable and incremental analysis of source code by leveraging on existing tools, and produce an output that conforms to the data that can be processed by GrimoireLab. Graal already offers analysis about code complexity, quality, dependencies, security and licensing, however currently it is not integrated with GrimoireLab.
+
+This idea is about adding support to GrimoireLab to produce source code related metrics using Graal.
 
 The aims of the project are as follows:
-  * ...
+  * Understanding the GrimoireLab components (Perceval, ELK, Mordred and Sigils) and the corresponding tool-chain.
+  * Adapting ELK and Mordred to be able to execute Graal and process the data produced.
+  * Producing analytics with Graal data and including them in Sigils.
+  * Evaluating the implementation with projects of different sizes.
+
+Other aims, such as enhancing Graal to support more analysis or improve existing ones are completely within scope.
 
 
 The aims may require [specific tasks].
```

Comment:
line 22: do you have a `specific task` that may be required to complete this project? This could be "extending GrimoireLab functionality to integrate Graal" or something.